src/_pytest/nodes.py

```diff
@@ -106,6 +127,10 @@ def __init__(
             if self.name != "()":
                 self._nodeid += "::" + self.name
 
+    @classmethod
+    def from_parent(cls, parent, **kw):
+        return cls._create(**kw, parent=parent)
+
```

Comment:
What's the benefit compared to making `parent` a required argument in general?  (would avoid a lot of changes at least)

Reply:
i need to establish a distinct construction pattern in order to then follow up with cleaning up the node construction (thats overly smart and auto-creating different things)

i tried a few times before without that and just failed flat on those occasions

Reply:
Ok.
But maybe then we could just revert it to "normal" object instantiation afterwards?  (i.e. before merging, not requiring all the warnings etc).

I think raising TypeErrors and using type annotations in general could be enough?

Reply:
as soon  as the ctor situation is largely resolved and the pattern is established, the meta class can go